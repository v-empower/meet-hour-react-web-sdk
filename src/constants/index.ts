export const DEFAULT_DOMAIN = 'meethour.io';
