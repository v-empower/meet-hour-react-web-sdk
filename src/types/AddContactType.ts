export default interface AddContactType {
    firstname: string,
    lastname?: string,
    email: string,
    phone?: string,
    country_code?: string,
    image?: string,
    is_show_portal?: boolean
  }