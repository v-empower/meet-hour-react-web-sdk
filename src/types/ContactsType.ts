export default interface ContactsType {
    limit: number,
    page: number,
    exclude_hosts: number
  }