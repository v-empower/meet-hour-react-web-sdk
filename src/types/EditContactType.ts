export default interface EditContactType {
    contact_id: number,
    firstname: string,
    lastname: string,
    phone: string,
    country_code: string,
    email: string,
    image: string,
    is_show_portal: boolean
  }