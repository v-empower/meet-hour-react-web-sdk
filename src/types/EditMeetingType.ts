interface UserObjectType {
  first_name?: string;
  last_name?: string;
  email?: string;
}

export default interface EditMeeting {
  meeting_id: string;
  meeting_name?: string;
  agenda?: string;
  passcode?: string;
  meeting_date?: string;
  meeting_time?: string;
  meeting_meridiem?: string;
  duration_hr?: number;
  duration_min?: number;
  timezone?: string;
  is_recurring?: number;
  recurring_type?: string;
  repeat_interval?: number;
  endBy?: string;
  end_date_time?: string;
  instructions?: string;
  is_show_portal?: number;
  enable_pre_registration?: number;
  meeting_topic?: string;
  meeting_agenda?: string;
  options?: Array<string>;
  old_attend?:
    | Array<number>
    | Array<UserObjectType>
    | Array<number & UserObjectType>;
  attend?:
    | Array<number>
    | Array<UserObjectType>
    | Array<number & UserObjectType>;
  groups?: Array<number>;
  hostusers?:
    | Array<number>
    | Array<UserObjectType>
    | Array<number & UserObjectType>;
}

