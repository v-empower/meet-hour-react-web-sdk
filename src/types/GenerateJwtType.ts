export default interface GenerateJwtType {
    meeting_id: string
    contact_id?: number,
    ui_config?: {
      APP_NAME?: string
      DEFAULT_LOGO_URL?: string
      BRAND_WATERMARK_BACKGROUND?: string
      SHOW_MEET_HOUR_WATERMARK?: string
      MEET_HOUR_WATERMARK_LINK?: string
      MOBILE_APP_PROMO?: string
      MOBILE_DOWNLOAD_LINK_ANDROID?: string
      MOBILE_DOWNLOAD_LINK_IOS?: string
      NATIVE_APP_NAME?: string
      PROVIDER_NAME?: string
      ENABLE_MOBILE_BROWSER?: string
      APP_SCHEME?: string
      ANDROID_APP_PACKAGE?: string
      HIDE_DEEP_LINKING_LOGO?: string
      disablePrejoinHeader?: string
      disablePrejoinFooter?: string
      toolbar_buttons?: Array<string>
    },
    config?: {
      resolution?: number,
      videoheightideal?: number,
      videoheightmax?: number,
      videoheightmin?: number,
      videowidthideal?: number,
      videowidthmax?: number,
      videowidthmin?: number,
      startAudioMuted?: number,
      p2penabled?: boolean,
      fileRecordingsEnabled?: boolean,
      liveStreamingEnabled?: boolean,
      dropboxappKey?: string
      dropboxredirectURI?: string
      enableWelcomePage?: string
      enableClosePage?: string
      requireDisplayName?: string
      disableDeepLinking?: string
    }
  }