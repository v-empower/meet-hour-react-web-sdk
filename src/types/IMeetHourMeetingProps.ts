/* eslint-disable-next-line */
import { MeetHourMeeting } from '..';
import IMeetingProps from './IMeetingProps';

/**
 * The type of the React {@code Component} props of {@link MeetHourMeeting}.
 */
export default interface IMeetHourMeetingProps extends IMeetingProps {

    /**
     * The domain used to build the conference URL.
     */
    domain?: string;
}
