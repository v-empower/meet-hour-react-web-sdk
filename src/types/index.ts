import { MeetHourExternalAPI } from './MeetHourExternalAPI';
declare global {
    interface Window {
        MeetHourExternalAPI: MeetHourExternalAPI;
    }
}

export { MeetHourExternalAPI } from './MeetHourExternalAPI';
export { default as IMeetHourMeetingProps } from './IMeetHourMeetingProps';
export { default as IMeetHourExternalApi } from './IMeetHourExternalApi';

