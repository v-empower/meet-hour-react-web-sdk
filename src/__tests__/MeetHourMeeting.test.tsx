import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import { DEFAULT_DOMAIN } from '../constants';
import { MeetHourMeeting } from '../index';
import { IMeetHourExternalApi } from '../types';

/* eslint-disable no-undef */

describe('MeetHourMeeting component', () => {
    it('should render correctly', () => {
        const props = {
            domain: DEFAULT_DOMAIN,
            roomName: 'TestingMeetHourMeetingComponent',
            onApiReady: (externalApi: IMeetHourExternalApi) => {
                console.log(externalApi);
            }
        };

        const wrapper = shallow(<MeetHourMeeting {...props} />);
        const snapshot = toJson(wrapper);


        // Overwrite non-deterministic fields
        snapshot.children[0].props.id = 'test';
        snapshot.children[0].props.key = 'test';
        expect(snapshot).toMatchSnapshot();
    });

    it('should render iframe', () => {
        const props = {
            domain: DEFAULT_DOMAIN,
            roomName: 'TestingMeetIFrame',
            onApiReady: (externalApi: IMeetHourExternalApi) => {
                console.log(externalApi);
            }
        };
        const wrapper = shallow(<MeetHourMeeting {...props} />);
        const iframe = wrapper.find('iframe');

        expect(iframe).toBeTruthy();
    });
});
