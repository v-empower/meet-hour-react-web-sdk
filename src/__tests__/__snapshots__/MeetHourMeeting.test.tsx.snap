// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetHourMeeting component should render correctly 1`] = `
<Fragment>
  <div
    id="test"
    key="test"
  />
</Fragment>
`;
